Framework/Core/include/Framework/DataAllocator.h

```diff
@@ -298,7 +298,9 @@ class DataAllocator
 
         if constexpr (std::is_pointer<typename T::value_type>::value == false) {
           // vector of elements
-          memcpy(payloadMessage->GetData(), object.data(), sizeInBytes);
+          if (object.data()) {
```

Comment:
i would rather check sizeInBytes.
But actually this looks ridiculous, and I cannot really follow the discussions that claim it is UB.

Reply:
```suggestion
          if (object.data() && sizeInBytes) {
```

Reply:
Checking sizeInBytes only might still trigger some static checker since what is UB is really having a nullptr...

Reply:
I don't think so, sizeInBytes should be sufficient in principle. Since if that is non-zero, also the pointers passed in must be non-zero.

Reply:
I think a static checker might not know that `sizeInBytes != 0 -> object.data() != nullptr`.

Reply:
But then why should a static checker complain? I mean he would need to know that object.data() = nullptr without sizeInBytes == nullptr? And then, do we actually check our code with static checkers?